lib/game.rb

```diff
@@ -0,0 +1,127 @@
+require 'pry'
+class Game
+
+	attr_accessor :player_1, :player_2, :board, :winner
+
+
+  # Winning combinations to win a game of Tic Tac Toe
+	WIN_COMBINATIONS = [[0,1,2],[3,4,5],[6,7,8],[0,3,6],[1,4,7],[2,5,8],[0,4,8],[6,4,2]]
+
+  # Initialize the game with default players and board - defaults to a two human players game
+	def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+		@player_1 = player_1
+    player_1.player = "Player one" # Give the players a name to differentiate between them
+		@player_2 = player_2
+    player_2.player = "Player two" # Give the players a name to differentiate between them
+		@board = board
+		@winner = 0
+	end
+
+  # Whose turn is it - identify the current player by which turn the game is up to
+  # Player one goes first with the "X" token
+	def current_player
+		if board.turn_count.odd?
+			player_2  # If the number of turns is an odd number it is Player two with the "O" token
+		else
+			player_1  # If the number of turns is an even number it is Player one with the "X" token
+		end
+	end
+
+  # Is the game over - if the board is full there are no move valid moves so the game is over
+  # If the game is drawn or won the game is over
+	def over?
+		true if board.full? || draw? || won?
+	end
+
+  # Is the game a draw - the board is full and no one has won (no one has gotten one of the winning combinations)
+	def draw?
+		true if board.full? && !won?
+	end
+
+  # Has a player won the game?
+	def won?
+		count_x = 0
```

Comment:
Again, I need more focus on return values! I needed to refactor #winner to incorporate your suggestion which in turn made my revisit and tighten my logic for this method.